app/controllers/api/v2/exchange_rates/period_lists_controller.rb

```diff
@@ -20,12 +20,26 @@ def serialized_period_list
         end
 
         def period_list
-          @period_list ||= ::ExchangeRates::PeriodList.build(year)
+          @period_list ||= ::ExchangeRates::PeriodList.build(year, type)
         end
 
         def year
           (params[:year].presence || ExchangeRateCurrencyRate.max_year).to_i
         end
+
+        def filter_params
+          params.fetch(:filter, {}).permit(:type)
+        end
+
+        def type
+          filter_params[:type]
+        end
+
+        def type
+          type_param = filter_params[:type]
+
+          type_param if [ExchangeRateCurrencyRate::SCHEDULED_RATE_TYPE, ExchangeRateCurrencyRate::SPOT_RATE_TYPE, ExchangeRateCurrencyRate::AVERAGE_RATE_TYPE].include?(type_param)
+        end
```

Comment:
Good that you are checking the type here!
We could be more explicit if you like:

```
            type_param = filter_params[:type]

          return nill unless [ExchangeRateCurrencyRate::SCHEDULED_RATE_TYPE, ExchangeRateCurrencyRate::SPOT_RATE_TYPE, ExchangeRateCurrencyRate::AVERAGE_RATE_TYPE].include?(type_param)
          
          type_param
        end
```